chain-signatures/node/src/storage/presignature_storage.rs

```diff
@@ -141,6 +140,19 @@ impl PresignatureStorage {
             .unwrap_or_default()
     }
 
+    pub async fn fetch_participants(&self, id: PresignatureId) -> Vec<Participant> {
```

Comment:
Keeping holders separately from Ps and Ts can create race conditions.

Reply:
im not sure what you mean by race conditions here? Can you give a couple examples of such cases?

Reply:
Can it happen that we will store the per-signature,  but not the holders and interpret this as nobody holds it except us?